### ISSUE TITLE
Handle Gradle dependencies in modules

### ISSUE THREAD

**Issue by DamnClin** [jhipster/jhipster-lite#1862]
Following https://github.com/jhipster/jhipster-lite/issues/1787 we can add gradle dependencies handling.

TODO: 

- [x] Update `FileSystemCurrentJavaDependenciesVersionsRepository` to have a strategy to handle either maven or gradle
- [x] Create a `JavaDependenciesCommandHandler` interface based on `MavenCommandHandler`
- [x] Create a `GradleCommandHandler` (which is a `JavaDependenciesCommandHandler`)
- [x] Update `FileSystemJavaDependenciesCommandsHandler` to create the most suited handler (depending on project version management)

**Comment by atomfrede** [comment]
@DamnClin I have created a first draft PR, maybe you can have a short look if thats heading in the right direction https://github.com/jhipster/jhipster-lite/pull/2748

**Comment by murdos** [comment]
@atomfrede : do you still intend to work on this feature? otherwise I would be happy to give it a try :-)

**Comment by atomfrede** [comment]
Feel to go on. I did not much so far. I think I would have tried rewrite to handle e.g. the dependency part.

**Comment by pascalgrimaud** [comment]
@murdos : if possible, try to split into small PRs or small tickets, as it will be easier to review and to add small bounties on each ;)

**Comment by murdos** [comment]
> @murdos : if possible, try to split into small PRs or small tickets, as it will be easier to review and to add small bounties on each ;)

My current roadmap is:
- [ ] introduce version catalogs feature with TOML in gradle build tool module: it seems to be the good practice, and will ease simplify manipulation of gradle build files that are hard to parse (I was thinking about using needles for that)
- [ ] Introduce GradleCommandHandler and JavaDependenciesCommandHandler interface, and implements the handling of the `SetVersion` command
- [ ] Handle other dependencies related commands (`(Add|Remove)DirectJavaDependency` and `(Add|Remove)JavaDependencyManagement`)
- [ ] Adapt CI tests and scripts to allow testing gradle projects
- [ ] Handle plugins related commands, but probably in another issue, since it will probably requires changing API related to build plugins in module builder ([related to this comment](https://github.com/jhipster/jhipster-lite/issues/4153#issuecomment-1366408863) and to the fact that plugin in gradle only have an id and a version - no groupId and artifactId)
- [ ] Stop hidding gradle build tool resources slug! :-)

**Comment by atomfrede** [comment]
@murdos Did you do already something with regards to dependency handling? If not I will try to have a look at the open rewrite approach. They have recipes for all things we need, so maybe we can use that.

**Comment by murdos** [comment]
I have work-in-progress using needles. 
I'm a bit skeptical about using openrewrite for this need, as IMO this would the setup and overhead would be overkill, but feel free to give it a try :-)

**Comment by atomfrede** [comment]
I will try. From the documentation it look quite easy to call different recipes from java code. Will keep you updated when I have something to show and discuss.

**Comment by atomfrede** [comment]
First thing I notice rewrite recipes only work for groovy based `build.gradle` not the kotlin dsl. I did the `.kts` decision a few month ago but not sure we really benefit from it. Anyways, I will to a small poc for manipulating `build.gradle` with rewrite via code outside of jhipster first.

**Comment by atomfrede** [comment]
I have a small sample using openrewrite. The idea would be to have a temp. `rewrite.yml` file which adds or removes dependencies and the rewrite step is executed via gradle + init script (which is also generated on the fly). But sadly, the gradle kotlin dsl is not supported yet, see here https://github.com/openrewrite/rewrite/issues/3291

The sample can be found here https://github.com/atomfrede/open-rewrite-for-jhipster-lite-sample Its a standard maven project which adds dependencies to an "external" gradle project.

**Comment by murdos** [comment]
Bounty claimed: https://opencollective.com/generator-jhipster/expenses/159351

**Comment by pascalgrimaud** [comment]
@murdos : approved, thanks for your work
